src/device/virtio/block/device.rs

```diff
@@ -136,32 +138,32 @@ where
 
 // We now implement `WithVirtioConfig` and `WithDeviceOps` to get the automatic implementation
 // for `VirtioDevice`.
-impl<M: GuestAddressSpace + Clone + Send + 'static> WithVirtioConfig<M> for Block<M> {
+impl<M: GuestAddressSpace + Clone + Send + 'static> VirtioDeviceType for Block<M> {
     fn device_type(&self) -> u32 {
         BLOCK_DEVICE_ID
     }
+}
 
-    fn virtio_config(&self) -> &VirtioConfig<M> {
+impl<M: GuestAddressSpace + Clone + Send + 'static> Borrow<VirtioConfig<M>> for Block<M> {
+    fn borrow(&self) -> &VirtioConfig<M> {
         &self.virtio_cfg
     }
+}
 
-    fn virtio_config_mut(&mut self) -> &mut VirtioConfig<M> {
+impl<M: GuestAddressSpace + Clone + Send + 'static> BorrowMut<VirtioConfig<M>> for Block<M> {
+    fn borrow_mut(&mut self) -> &mut VirtioConfig<M> {
         &mut self.virtio_cfg
     }
 }
 
-impl<M: GuestAddressSpace + Clone + Send + 'static> WithDeviceOps for Block<M> {
+impl<M: GuestAddressSpace + Clone + Send + 'static> VirtioDeviceActions for Block<M> {
     type E = Error;
 
     fn activate(&mut self) -> Result<()> {
         if self.virtio_cfg.device_activated {
             return Err(Error::AlreadyActivated);
         }
 
-        if !self.queues_valid() {
```

Comment:
And this one. Please check!

Reply:
should be fine. But now that all dependent crates are updated, this code probably doesn't work anyways.